### PR TITLE
Upgrade the DataLogging module and allow it to split files as it writes and generate less code

### PR DESCRIPTION
**Pull request by TrentHouliston**

### CODE REVIEW

File: module/support/logging/DataLogging/src/DataLogging.cpp [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                            }
+                        }
+
+                        // Unbind any recorders we didn't save
+                        for (auto& handle : handles) {
+                            log<NUClear::INFO>("Data logging for type", handle.first, "disabled");
+                            handle.second.unbind();
+                        }
+
+                        // Add any recorders new ones that we don't have yet
```
Comment (Bidski) [inline module/support/logging/DataLogging/src/DataLogging.cpp]: You what now?

File: module/support/logging/DataLogging/src/DataLogging.cpp [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+                        for (auto& setting : cfg["messages"].config) {
+                            std::string name = setting.first.as<std::string>();
+                            bool enabled     = setting.second.as<bool>();
+
+                            // If it was enabled and we are keeping it enabled, keep it and remove it from the old list
```
Comment (Bidski) [inline module/support/logging/DataLogging/src/DataLogging.cpp]: You sure that is what this does?

File: shared/utility/support/math_string.cpp [hunk 3]
```diff
@@ -40,6 +40,11 @@ namespace support {
         table.add_constants();
         table.add_constant("auto", std::numeric_limits<double>::infinity());
 
+        // File size constants
+        table.add_constant("KiB", 1024);
+        table.add_constant("MiB", 1048576);
+        table.add_constant("GiB", 1073741824);
```
Comment (Bidski) [inline shared/utility/support/math_string.cpp]: Is it worth replacing the raw values with `std::pow(2, 10)`, etc? Or maybe as a [hex floating point literal](https://en.cppreference.com/w/cpp/language/floating_literal)?
```
0x1p10 = 1024
0x1p20 = 1048576
0x1p30 = 1073741824
```

File: module/support/logging/DataLogging/src/DataLogging.cpp (hunk 1 above)
Comment (TrentHouliston) [reply module/support/logging/DataLogging/src/DataLogging.cpp]: I speak well the english!

File: shared/utility/support/math_string.cpp (hunk 3 above)
Comment (TrentHouliston) [reply shared/utility/support/math_string.cpp]: Maybe, but these are pretty safe

File: module/support/logging/DataLogging/src/DataLogging.cpp (hunk 2 above)
Comment (TrentHouliston) [reply module/support/logging/DataLogging/src/DataLogging.cpp]: I am sure that this is not what it does? That's kinda the same...

**Pull request merged**